jax/experimental/jax2tf/jax2tf.py

```diff
@@ -46,6 +47,17 @@
 from jaxlib import xla_client
 
 
+# TF scope validation, refer to tensorflow/python/framework/ops.py
```

Comment:
At a guess this xref could get out of date quickly.. Is there a more canonical place (e.g. documentation for what a valid name scope is?). The `ops.py` implementation also seems to be branched from `node_def_util.cc:737`. Should we link to that?

Also I've spotted some similar code in other projects, is this (sanitizing a name scope) something we should ask TF team to support (e.g. `tf.name_scope(..)` would do this for you?).

Reply:
Replaced with https://github.com/tensorflow/tensorflow/blob/r2.3/tensorflow/core/framework/node_def_util.cc#L731 which should be stable.

I consulted with TF team member who touched the code over a year ago, the name scope code in TF is quite old and it would be hard to introduce new behaviors.  The new hidden behavior might cause a bit of confusion, and the backward compatibility could also be an issue. It doesn't look too promising for TF team to support this.